Expand RFC Pattern (#245)
gruetter/innerSourceTest: patterns/2-structured/transparent-cross-team-decision-making-using-rfcs.md

- **democratization of the decision making process** for decisions that impact many teams (also offloading team leads from that burden)
- **a open asynchronous communication method** that works well across multiple teams and geos
- **empowers individuals and teams** to effect large scale change
- **record of decisions made** for people to refer back to for context
- **scales impact of experienced engineers** as they can contribute to solutions asynchronously and remotely, rather than needing to be present in a meeting
- **terminology alignment** e.g. by spelling out our testing terminology such as “what is a system test?”
- **process alignment** e.g. by spelling out the out-of-hours support process
- **greater clarity of thought**, as writing an RFC makes the author challenge herself more than they would normally

The RFC approach also has risk that we want to call out:

- It doesn’t always work! e.g. some people might still argue against a decision that was already made via an RFC. However having the decision making process in writing is still beneficial in these scenarios, as you can point people to when and why a certain decision was made.

> - Writing up design proposals (architecture, protocols, etc) upfront has an element of upfront waterfall-like design to it that does not fit the iterative development approach that many development teams favor. Remember: “Working software over comprehensive documentation” (Agile Manifesto). The RFC process should be as lightweight as possible.

Reviewer: ```suggestion
- Writing up design proposals (architecture, protocols, etc) upfront has an element of waterfall-like design that does not fit the iterative development approach that many development teams favor. Remember: “Working software over comprehensive documentation” ([Agile Manifesto](https://agilemanifesto.org/)). The RFC process should be as lightweight as possible.
```

Adding a link. Many people will already know the manifesto but hey, doesn't hurt :)

Also shortening the text a bit.

Author: Good suggestion :+1:

> - **Uber** - According to this blog post by Gergely Orosz: [Scaling Engineering Teams via Writing Things Down and Sharing - aka RFCs][uber].
> - **Europace** - As described in Open Organization: [Setting cross-team standards and best practices in the open][europace].
> - **BBC iPlayer & Sounds** - As presented at the ISC Fall Summit 2020 [Using Internal RFCs to Enhance Collaboration][bbc].
> - **Google Design Docs** - As described in this blog post by Malte Ubl [Design Docs at Google][google]

Reviewer: ```suggestion
- **BBC iPlayer & Sounds** - As presented at the ISC Fall Summit 2020 [Using Internal RFCs to Enhance Collaboration][bbc].
- **Europace** - As described in Open Organization: [Setting cross-team standards and best practices in the open][europace].
- **Uber** - According to this blog post by Gergely Orosz: [Scaling Engineering Teams via Writing Things Down and Sharing - aka RFCs][uber].
- **Google Design Docs** - As described in this blog post by Malte Ubl [Design Docs at Google][google]
```

I vote for calling out the BBC first, as you are the one putting in the work to contribute this pattern.

Author: Sure, happy to take this change :)

## Status

**Structured**

## Author(s)

- Tom Sadler
- Sebastian Spier

> # Aliases

Reviewer: ```suggestion
## Aliases
```

Reducing level of heading.

Reviewer: This should be non-controversial, so I am committing this suggestion.

- [Design Docs][google]

> - Architecture Decision Record (ADRs)
>     - Not necessarily a direct alias, as they can sometimes be used very differently e.g. RFCs for seeking input and building consensus, ADRs for recording decisions and implementation detail

Reviewer: ```suggestion
- Architecture Decision Record (ADRs) - Not necessarily a direct alias, as they can sometimes be used very differently e.g. RFCs for seeking input and building consensus, ADRs for recording decisions and implementation detail
```

Simplifying the list.

> 

Reviewer: ```suggestion
```

Removing one newline to make the linter happy.
